testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/RuntimeJournalTypeManagementTestCase.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2011, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.manualmode.messaging;
+
+import static org.jboss.as.controller.client.helpers.ClientConstants.VALUE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+import org.jboss.arquillian.container.test.api.ContainerController;
+
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test reading the journal runtime type.
+ *
+ * @author Emmanuel Hugonnet (c) 2019 Red Hat, Inc.
+ */
+@RunAsClient()
+@RunWith(Arquillian.class)
+public class RuntimeJournalTypeManagementTestCase {
+
+    private static final String DEFAULT_FULL_JBOSSAS = "default-full-jbossas";
+
+    private static final ModelNode SERVER_ADDRESS = PathAddress.parseCLIStyleAddress("/subsystem=messaging-activemq/server=default").toModelNode();
+
+    @ArquillianResource
+    protected static ContainerController container;
+
+    @Test
+    public void testReadFileJournaltype() throws IOException {
+        if (!container.isStarted(DEFAULT_FULL_JBOSSAS)) {
+            container.start(DEFAULT_FULL_JBOSSAS);
+        }
+        ManagementClient managementClient = createManagementClient();
+        assertEquals("ASYNCIO", readJournalType(managementClient));
+        assertNotNull(readRuntimeJournalType(managementClient));
```

Comment:
Is there a reason why we only check this is not null?
What about `assertEquals("ASYNCIO", readRuntimeJournalType(managementClient));` instead ?
Before falling back to NIO, these attributes should be the same, right?

Reply:
because we can't presume of the value as on Windows it would always be NIO for example